bin/copyRawDataToTmp.sh

```diff
@@ -0,0 +1,356 @@
+# * sourcing config files,
+# * and parsing commandline arguments,
+# but before doing the actual data transfers.
+#
+# As servernames and folders may contain various characters that would require escaping in (lock) file names,
+# we compute a hash for the combination of ${samplesheetsServerLocation} and ${samplesheetsLocation} to append to the ${SCRIPT_NAME}
+# for creating unique lock file. We write the combination of ${samplesheetsServerLocation} and ${samplesheetsLocation} in the lock file
+# to make it easier to detect which combination of ${samplesheetsServerLocation}and ${samplesheetsLocation} the lock file is for.
+#
+hashedSource="$(printf '%s:%s' "${samplesheetsServerLocation}" "${samplesheetsLocation}" | md5sum | awk '{print $1}')"
+lockFile="${WORKING_DIR}/logs/${SCRIPT_NAME}_${hashedSource}.lock"
+thereShallBeOnlyOne "${lockFile}"
+printf 'Lock file for %s instance that pushes data to %s:%s\n' "${SCRIPT_NAME}" "${samplesheetsServerLocation}" "${samplesheetsLocation}" > "${lockFile}"
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Successfully got exclusive access to lock file ${lockFile} ..."
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Log files will be written to ${WORKING_DIR}/logs ..."
+
+#
+# Use multiplexing to reduce the amount of SSH connections created
+# when rsyncing using the group's data manager account.
+# 
+#  1. Become the "${DATA_MANAGER} user who will rsync the data to prm and 
+#  2. Add to ~/.ssh/config:
+#		ControlMaster auto
+#		ControlPath ~/.ssh/tmp/%h_%p_%r
+#		ControlPersist 5m
+#  3. Create ~/.ssh/tmp dir:
+#		mkdir -p -m 700 ~/.ssh/tmp
+#  4. Recursively restrict access to the ~/.ssh dir to allow only the owner/user:
+#		chmod -R go-rwx ~/.ssh
+#
+
+#
+# Get a list of all samplesheets for this group on the specified samplesheetsServerLocation, where the raw data was generated, and
+#	1. Loop over their analysis ("run") sub dirs and check if there are any we need to rsync.
+#	2. Optionally, split the samplesheets per project after the data was rsynced.
+#
+# shellcheck disable=SC2029
+readarray -t sampleSheetsFolder < <(ssh "${samplesheetsServerLocation}" "find \"${samplesheetsLocation}/\" -mindepth 1 -maxdepth 1 -type f -name '*.${SAMPLESHEET_EXT}'")
+
+##ISSUE: this script should always be executed by the ateambot user (and not for chaperone with the dm user solely to write the logs)
+if [[ "${#sampleSheetsFolder[@]:-0}" -eq '0' ]]
+then
+	log4Bash 'WARN' "${LINENO}" "${FUNCNAME:-main}" '0' "No samplesheets found at ${samplesheetsServerLocation} ${samplesheetsLocation}/*.${SAMPLESHEET_EXT}."
+else
+	for sampleSheet in "${sampleSheetsFolder[@]}"
+	do
+		#
+		# Process this samplesheet / run 
+		#
+		project="$(basename "${sampleSheet%."${SAMPLESHEET_EXT}"}")"
+		controlFileBase="${WORKING_DIR}/logs/${project}/"
+		export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${project}.${SCRIPT_NAME}"
+		printf '%s\n' "Rsyncing ${project} dir ..." >> "${lockFile}"
```

Comment:
```suggestion
		printf '%s\n' "Processing ${project} dir ..." >> "${lockFile}"
```
Because we first check if `${JOB_CONTROLE_FILE_BASE}.finished` exists and if true will skip rsync for this project.